app/models/piece.rb

```diff
@@ -2,8 +2,58 @@ class Piece < ActiveRecord::Base
   belongs_to :users
   belongs_to :games
 
-def is_obstructed?
-	return false
-end
+	def is_obstructed?(x, y)
+
+		#check moving direction
+		path = moving_direction(x, y)
+
+		#path is horizontal from left to right
+		if path == 'horizontal' && x_cord < x
+			(x_cord + 1).upto(x - 1) do |changing_x|
```

Comment:
What happens if x_cord = 0 and x = 1 and it's implemented with the .upto callback? Will the .upto callback work with `(0 + 1).upto(1 - 1)`?